geode-gfsh/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/OfflineDiskStoreCommandsDUnitTest.java

```diff
@@ -0,0 +1,244 @@
+    server.invoke(() -> {
+      createServerWithRegionAndPersistentRegion(diskStoreDirectories);
+      Region<String, String> region = cacheRule.getCache().getRegion(REGION_NAME);
+      IntStream.range(0, ENTRIES).forEach(i -> region.put("Key_" + i, "Value_" + i));
+    });
+    locator.invoke(this::gracefullyDisconnect);
+    server.invoke(this::gracefullyDisconnect);
+    gfsh.executeAndAssertThat(
+        baseCommand + " --name=" + DISK_STORE_ID + " --disk-dirs=" + diskDirs)
+        .statusIsSuccess();
+  }
+
+  @Test
+  @Parameters({"describe offline-disk-store",
+      "validate offline-disk-store",
+      "alter disk-store --region=testRegion --enable-statistics=true"})
+  public void testThreadHangWithOfflineDiskStoreCommands(String baseCommand) throws IOException {
+    VM locator = getVM(0);
+    VM server = getVM(1);
+    final int ENTRIES = 100000;
+    int site1Port = getRandomAvailableTCPPortsForDUnitSite(1)[0];
+    String threadName = "Asynchronous disk writer for region";
+    int counter = 0;
+
+    File diskStoreDirectory1 = temporaryFolder.newFolder("diskDir1");
+
+    File[] diskStoreDirectories =
+        new File[] {diskStoreDirectory1};
+    String diskDirs = Arrays.stream(diskStoreDirectories).map(File::getAbsolutePath)
+        .collect(Collectors.joining(","));
+
+    locator.invoke(() -> cacheRule.createCache(createLocatorConfiguration(site1Port)));
+    server.invoke(() -> cacheRule.createCache(createServerConfiguration(site1Port)));
+    server.invoke(() -> {
+      createServerWithRegionAndPersistentRegion(diskStoreDirectories);
+      Region<String, String> region = cacheRule.getCache().getRegion(REGION_NAME);
+      IntStream.range(0, ENTRIES).forEach(i -> region.put("Key_" + i, "Value_" + i));
+    });
+    locator.invoke(this::gracefullyDisconnect);
+    server.invoke(this::gracefullyDisconnect);
+
+    gfsh.executeAndAssertThat(
+        baseCommand + " --name=" + DISK_STORE_ID + " --disk-dirs=" + diskDirs)
+        .statusIsSuccess();
+
+    File tempFile = temporaryFolder.newFile("dumpFile.txt");
+    BufferedWriter writer = new BufferedWriter(new FileWriter(tempFile));
+    ThreadMXBean bean = ManagementFactory.getThreadMXBean();
+    ThreadInfo[] infos = bean.dumpAllThreads(true, true);
+    for (ThreadInfo info : infos) {
+      if (info.toString().contains(threadName))
+        writer.append(info.toString());
+    }
+
+    writer.close();
+
+    try (BufferedReader br = new BufferedReader(new FileReader(tempFile))) {
+      String line;
+      while ((line = br.readLine()) != null) {
+        if (line.contains(threadName))
+          counter++;
+      }
+    }
+    assertThat(counter).isEqualTo(0);
+  }
+
+  @Test
+  @Parameters({"compact offline-disk-store", "describe offline-disk-store",
+      "validate offline-disk-store",
+      "alter disk-store --region=testRegion --enable-statistics=true"})
+  public void testOfflineCommandsWithMultipleDirsAndWrongName(String baseCommand)
```

Comment:
Can you change the test name to something more meaningful?, maybe `offlineDiskStoreCommandShouldFailWhenDiskStoreFileDoesNotExist`?.